msg/messages/message_context.go

```diff
@@ -0,0 +1,54 @@
+package messages
```

Comment:
maybe this package named recorder more suitable

Reply:
I think `messages` is OK.